R/lazy-join-query.R

```diff
@@ -245,10 +245,19 @@ sql_build.lazy_rf_join_query <- function(op, con, ...) {
   by$x_as <- ident(table_names_out[[1]])
   by$y_as <- ident(table_names_out[[2]])
 
+  select <- sql_rf_join_vars(
+    con,
+    type = op$type,
+    vars = vars_classic,
+    x_as = by$x_as,
+    y_as = by$y_as
+  )
+
   join_query(
     sql_optimise(sql_build(op$x, con), con),
     sql_optimise(sql_build(op$y, con), con),
     vars = vars_classic,
+    select = select,
```

Comment:
Theoretically, we could remove `vars` but this would be a breaking change. But it might also be quite hard to properly deprecate `vars`. Should we just accept the breaking change? I don't expect many methods for `sql_render.join_query()` apart from the ones in dbplyr.

Reply:
I think it's probably fine to just break it, but you might want to double check affected packages with this search: https://github.com/search?type=code&auto_enroll=true&q=org%3Acran+sql_render+path%3ANAMESPACE.